packages/terra-menu/src/Menu.jsx

```diff
@@ -0,0 +1,214 @@
+  /**
+   * List of Menu.Item(s)/Menu.ItemGroup(s)/Menu.Divider(s) to be displayed as content within the Menu.
+   */
+  children: PropTypes.node.isRequired,
+  /**
+   * Callback function indicating a close condition was met, should be combined with isOpen for state management.
+   */
+  onRequestClose: PropTypes.func.isRequired,
+  /**
+   * Target element for the menu to anchor to.
+   */
+  targetRef: PropTypes.func.isRequired,
+  /**
+   * Bounding container for the menu, will use window if no value provided.
+   */
+  boundingRef: PropTypes.func,
+  /**
+   * CSS classnames that are append to the arrow.
+   */
+  classNameArrow: PropTypes.string,
+  /**
+   * CSS classnames that are append to the menu content inner.
+   */
+  classNameContent: PropTypes.string,
+  /**
+   * CSS classnames that are append to the overlay.
+   */
+  classNameOverlay: PropTypes.string,
+  /**
+   * Should the menu be presented as open.
+   */
+  isOpen: PropTypes.bool,
+  /**
+   * A string representation of the width in px, limited to:
+   * 160, 240, 320, 640, 960, 1280, 1760
+   */
+  contentWidth: PropTypes.oneOf(Object.keys(MenuWidths)),
+  /**
+   * Indicates if the menu should have an center aligned arrow displayed on dropdown.
+   * Otherwise, the menu will display without an arrow and right aligned.
+   */
+  isArrowDisplayed: PropTypes.bool,
+};
+
+const defaultProps = {
+  isArrowDisplayed: false,
+  children: [],
+  isOpen: false,
+  contentWidth: '240',
+};
+
+const MENU_PADDING_TOP = 6;
+const MENU_PADDING_BOTTOM = 6;
+const MENU_ITEM_HEIGHT = 23;
+const MENU_DIVIDER_HEIGHT = 10;
+
+class Menu extends React.Component {
+  constructor(props) {
+    super(props);
+    this.getContentHeight = this.getContentHeight.bind(this);
+    this.getInitialState = this.getInitialState.bind(this);
+    this.push = this.push.bind(this);
+    this.pop = this.pop.bind(this);
+    this.state = this.getInitialState();
+  }
+
+  getInitialState() {
+    const initialMenu = (
+      <MenuContent
+        key="MenuPage-0"
+        onRequestNext={this.push}
+        onRequestBack={this.pop}
+        onRequestClose={this.props.onRequestClose}
+        index={0}
+      >
+        {this.props.children}
+      </MenuContent>
+    );
+
+    return {
+      stack: [initialMenu],
+    };
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (this.props.isOpen && !nextProps.isOpen) {
+      this.setState(this.getInitialState());
+    }
+  }
+
+  getContentHeight() {
+    let itemCount = 0;
+    let dividerCount = 0;
+
+    React.Children.forEach(this.props.children, (child) => {
+      if (child.props.children && child.props.children.length > 0) {
+        itemCount += child.props.children.length;
+      } else if (child.type === <MenuDivider />.type) {
+        dividerCount += 1;
+      } else {
+        itemCount += 1;
+      }
+    });
+
+    return (itemCount * MENU_ITEM_HEIGHT) + (dividerCount * MENU_DIVIDER_HEIGHT) + MENU_PADDING_TOP + MENU_PADDING_BOTTOM;
```

Comment:
@mhemesath The fixed sizes for popups prevent the need for multiple renders and multiple re-positions when dealing with popups bounded by screen/bounding rects.
There are also technical limitations when used with headers/flex layouts. If the consumer wants a header, and we are changing the size of its parent container when constrained, it will break the content’s scrolling. 

The general thought from UX was that the content in the popup should be responsive, rather than the popup respond to the content. It also gives UX a common toolset to leverage for consistent popup sizing.

To summarize, if the consumer wants the popup to by dynamically sized:
 - It will be forced to render multiple times
 - Popup content cannot use flex positioning or sizing using percentages
 - If consumers provide their own set content size to allow for flex/percentage sizing, we cannot realistically manage bounding conflicts.